go-controller/pkg/ovn/egressgw.go

```diff
@@ -18,13 +18,23 @@ import (
 	"k8s.io/klog/v2"
 )
 
+type gatewayInfo struct {
```

Comment:
just curious why we need a separate struct. can't we just pass the boolean around to methods that need bfd based on the annotation?

Reply:
That was my initial version, but then I realized that the gws are also referenced in the nsInfo structure (see https://github.com/ovn-org/ovn-kubernetes/pull/2077/files#diff-796adb9d8a9b5b5636dc81e7350a00ef143e6e5bcb52ec30dc350c866cd9208cL98 and https://github.com/ovn-org/ovn-kubernetes/pull/2077/files#diff-796adb9d8a9b5b5636dc81e7350a00ef143e6e5bcb52ec30dc350c866cd9208cL106 , where it's contained as map entry). 
Not having a structure would have made the whole thing hard to follow. Each (set of) gws have this property which is bfdEnabled or not, which makes it natural to have it wrapped in its own type.